cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 The Cobalt Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for scroll containers."""
+
+import logging
+
+from cobalt.black_box_tests import black_box_tests
+from cobalt.black_box_tests.threaded_web_server import ThreadedWebServer
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.common import keys
+
+# Time to sleep after a mouse move to give the device time to process it and
+# to avoid collapsing with subsequent move events.
+_SLEEP_AFTER_MOVE_TIME = 0.5
+
+
+def find_element_by_id(runner, id_selector):
+  return runner.webdriver.find_elements_by_css_selector('#' + id_selector)[0]
+
+
+class PointerTest(black_box_tests.BlackBoxTestCase):
+  """Tests pointer and mouse event."""
+
+  def test_pointer_events(self):
+    with ThreadedWebServer(binding_address=self.GetBindingAddress()) as server:
+      url = server.GetURL(
+          file_name='testdata/scroll_with_none_pointer_events.html')
+
+      with self.CreateCobaltRunner(url=url) as runner:
+        logging.info('JS Test Setup WaitForJSTestsSetup')
+        runner.WaitForJSTestsSetup()
+        logging.info('JS Test Setup')
+        self.assertTrue(runner.webdriver)
+
+        row = find_element_by_id(runner, 'row')
+        actions = ActionChains(runner.webdriver)
+
+        # No scrolling when pointer-events:none
+        actions.click_and_hold(row).pause(_SLEEP_AFTER_MOVE_TIME)
+        actions.move_by_offset(-100, 0)
+        actions.release()
+        actions.perform()
+        runner.SendKeys(keys.Keys.NUMPAD0)
```

Comment:
I thought I had posted this already, but I don't see it—apologies if I'm reposting.

If we call `onEndTest` when we send `NUMPAD0`, won't we not run the second test case below?

Reply:
It will still run the second test case. See the definition of this function: https://github.com/youtube/cobalt/blob/3da73c7c6f9b04beae664905608493abd32bcfe1/cobalt/black_box_tests/testdata/black_box_js_test_utils.js#L118

This function won't really end the test, but only set the test result. If there are no previous failures, it will set the test result as success. In short, the test still behaves correctly, but might be a little bit confusing. I'll make the change to only call this at the end to make it less confusing.

Reply:
Ah, thanks for the clarification. What do you think about renaming this function (not in this PR, but in general)?

Reply:
Yeah, I agree this is confusing, and I think people are using this function as to end test instead of just setting the end state of the test.  I'm not sure shall we rename this function and indicate this function doesn't end the test, or shall we actually change this function to end the test. More investigation are needed then simply renaming it. May worth a follow-up.